### PR TITLE
Feat(eos_designs): Add support for TerminAttr token-secure auth for on-premise CV

### LINKED ISSUE #2025: Feat(eos_designs): Add support for Terminattr token/cert based auth
### Enhancement summary

Today we support only key-based auth for Terminattr towards on-prem.
For CVaaS we support token.

Since this will be breaking we cannot change the default behavior in 3.x, but we should definitely use token by default from AVD 4.x. For now we will need to have a knob.

### Which component of AVD is impacted

eos_designs

### Use case example

Newer releases of CVP will only support token.

### Describe the solution you would like

N/A

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Add support for TerminAttr token-secure auth for on-premise CV

## Related Issue(s)

Fixes #2025 

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

- Sneaking a few fixes for porting guide in here
- Changing behavior if ingestkey is not set, to instead configure token auth.
  This is non breaking for any deployments, since they would have the key set already.
  CV will deprecate support for key based auth soon, so we need to move people towards
  cert based auth also for on-prem.

  ```yaml
  On-premise CVP ingest auth key. If set, terminattr will be configured with key-based authentication for on-premise CVP.
  If not set, terminattr will be configured with certificate based authentication using token-secure onboarding.
  Token must be copied to the device first.
  ```

## How to test

Added molecule test without key.
No changes to other molecule scenarios having keys set.

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_designs/docs/Management Settings.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ search:
 
     | Variable | Type | Required | Default | Value Restrictions | Description |
     | -------- | ---- | -------- | ------- | ------------------ | ----------- |
-    | [<samp>cvp_ingestauth_key</samp>](## "cvp_ingestauth_key") | String |  |  |  | CloudVision ingest authentication key is required for on-prem CVP |
+    | [<samp>cvp_ingestauth_key</samp>](## "cvp_ingestauth_key") | String |  |  |  | On-premise CVP ingest auth key. If set, terminattr will be configured with key-based authentication for on-premise CVP.<br>If not set, terminattr will be configured with certificate based authentication using token-secure onboarding.<br>Token must be copied to the device first. |
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/docs/Management Settings.md]:
```suggestion
    | [<samp>cvp_ingestauth_key</samp>](## "cvp_ingestauth_key") | String |  |  |  | On-premise CVP ingest auth key. If set, TerminAttr will be configured with key-based authentication for on-premise CVP.<br>If not set, TerminAttr will be configured with certificate based authentication using token-secure onboarding.<br>Token must be copied to the device first. |
```

**Review by gmuloc — APPROVED** [review]
Only comment about making terminattr -> TerminAttr for comments / descriptions to be consistent with some other buts of the doc (in the schema as well)

**Review by carlbuchmann — APPROVED** [review]
LGTM

**Review by gmuloc — APPROVED** [review]

**Comment by noredistribution** [comment]
one quick comment here, `token-secure` method is only supported on CVaaS, whereas on `on-prem` it has to be `token`
they are essentially the same mechanism but `token-secure` will only work on CVaaS and `token` will only work on on-prem, so we will have to make sure to use `"method": "token"` unless CVaaS is used in which case it's `"method": "token-secure"

**Pull request merged**